overlord/snapstate/backend/copydata_test.go

```diff
@@ -582,6 +583,108 @@ func (s *copydataSuite) TestUndoCopyDataSameRevision(c *C) {
 	}
 }
 
+const (
+	mountRunMntUbuntuSaveFmt = `26 27 8:3 / %s/run/mnt/ubuntu-save rw,relatime shared:7 - ext4 /dev/fakedevice0p1 rw,data=ordered`
+	mountSnapSaveFmt         = `26 27 8:3 / %s/var/lib/snapd/save rw,relatime shared:7 - ext4 /dev/fakedevice0p1 rw,data=ordered`
+)
+
+func (s *copydataSuite) TestSetupCommonSaveDataClassic(c *C) {
+	v1 := snaptest.MockSnap(c, helloYaml1, &snap.SideInfo{Revision: snap.R(10)})
+
+	// first install
+	err := s.be.SetupSnapSaveData(v1, progress.Null)
+	c.Assert(err, IsNil)
+	_, err = os.Stat(v1.CommonDataSaveDir())
+	c.Assert(err.Error(), Equals, fmt.Sprintf("stat %s/var/lib/snapd/save/snap/hello: no such file or directory", dirs.GlobalRootDir))
+}
+
+func (s *copydataSuite) TestSetupCommonSaveDataCoreNoMount(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+	restore = osutil.MockMountInfo("")
+	defer restore()
+	v1 := snaptest.MockSnap(c, helloYaml1, &snap.SideInfo{Revision: snap.R(10)})
+
+	// first install
+	err := s.be.SetupSnapSaveData(v1, progress.Null)
+	c.Assert(err, IsNil)
+	_, err = os.Stat(v1.CommonDataSaveDir())
+	c.Assert(err.Error(), Equals, fmt.Sprintf("stat %s/var/lib/snapd/save/snap/hello: no such file or directory", dirs.GlobalRootDir))
+}
+
+func (s *copydataSuite) TestSetupCommonSaveDataFirstInstall(c *C) {
+	restore := release.MockOnClassic(false)
+	defer restore()
+	restore = osutil.MockMountInfo(fmt.Sprintf(mountRunMntUbuntuSaveFmt, dirs.GlobalRootDir) + "\n" +
+		fmt.Sprintf(mountSnapSaveFmt, dirs.GlobalRootDir))
+	defer restore()
+
+	v1 := snaptest.MockSnap(c, helloYaml1, &snap.SideInfo{Revision: snap.R(10)})
+
+	// first install
+	err := s.be.SetupSnapSaveData(v1, progress.Null)
+	c.Assert(err, IsNil)
+	_, err = os.Stat(v1.CommonDataSaveDir())
+	c.Assert(err, IsNil)
+
```

Comment:
At this point, please create a file under the save dir, to make sure that it gets removed later too.

Reply:
Thank you, you are right! Fixed